python/lsst/obs/subaru/instrument.py

```diff
@@ -216,48 +215,8 @@ def writeCuratedCalibrations(self, butler):
                 continue
             butler.put(entry, datasetType, {"instrument": self.getName()})
 
-        # Write defects with validity ranges taken from obs_subaru_data/hsc/defects
-        # (along with the defects themselves).
-        datasetType = DatasetType("defects", ("instrument", "detector", "calibration_label"), "Defects",
-                                  universe=butler.registry.dimensions)
-        butler.registry.registerDatasetType(datasetType)
-        defectPath = os.path.join(getPackageDir("obs_subaru_data"), "hsc", "defects")
-        camera = self.getCamera()
-        defectsDict = read_all(defectPath, camera)[0]  # This method returns a dict plus the calib type
-        endOfTime = TIMESPAN_MAX
-        dimensionRecords = []
-        datasetRecords = []
-        # First loop just gathers up the things we want to insert, so we
-        # can do some bulk inserts and minimize the time spent in transaction.
-        for det in defectsDict:
-            detector = camera[det]
-            times = sorted([k for k in defectsDict[det]])
-            defects = [defectsDict[det][time] for time in times]
-            times = [astropy.time.Time(t, format="datetime", scale="utc") for t in times]
-            times += [endOfTime]
-            for defect, beginTime, endTime in zip(defects, times[:-1], times[1:]):
-                md = defect.getMetadata()
-                calibrationLabel = f"defect/{md['CALIBDATE']}/{md['DETECTOR']}"
-                dataId = DataCoordinate.standardize(
-                    universe=butler.registry.dimensions,
-                    instrument=self.getName(),
-                    calibration_label=calibrationLabel,
-                    detector=detector.getId(),
-                )
-                datasetRecords.append((defect, dataId))
-                dimensionRecords.append({
-                    "instrument": self.getName(),
-                    "name": calibrationLabel,
-                    "datetime_begin": beginTime,
-                    "datetime_end": endTime,
-                })
-        # Second loop actually does the inserts and filesystem writes.
-        with butler.transaction():
-            butler.registry.insertDimensionData("calibration_label", *dimensionRecords)
-            # TODO: vectorize these puts, once butler APIs for that become
-            # available.
-            for defect, dataId in datasetRecords:
-                butler.put(defect, datasetType, dataId)
+        # Write defects
+        self.writeStandardTextCuratedCalibrations(butler)
```

Comment:
The comment is misleading, I think: it's writing more than just defects, it's whatever is in `standardCuratedDatasetTypes`.